pkg/driver/memory/byteslice.go

```diff
@@ -1,22 +1,24 @@
 package memory
 
 import (
+	"bytes"
 	"fmt"
 	"io"
+
+	"github.com/lf-edge/adam/pkg/driver/common"
 )
 
 type ByteSlice struct {
 	// we do this as a slice of byte slice, rather than a single byte slice,
 	// because we need to track breaks, so we can delete from the beginning
 	data         [][]byte
-	dataCache    []byte
 	currentRead  int
 	readComplete bool
 	maxSize      int
 	size         int
 }
 
-func (bs ByteSlice) Get(index int) ([]byte, error) {
+func (bs *ByteSlice) Get(index int) ([]byte, error) {
```

Comment:
Is this all converted to pointer because of the need for `ChunkReader` to track current state? Is there any other reason for it?

Reply:
Correct, to be able to adjust `currentRead` in `NextChunkReader`. And as I remember it is recommended way to sync all receivers in terms of name and signature.